src/plugins/progress/components/CustomBlock.vue

```diff
@@ -0,0 +1,551 @@
+import { getInstance } from '@snapshot-labs/lock/plugins/vue3';
+import { signMessage } from '@snapshot-labs/snapshot.js/src/utils/web3';
+import { ref, onMounted } from 'vue';
+import { useFlashNotification } from '@/composables/useFlashNotification';
+
+const { notify } = useFlashNotification();
+const { web3Account } = useWeb3();
+const props = defineProps([
+  'space',
+  'proposal',
+  'results',
+  'loaded',
+  'strategies'
+]);
+let loading = false;
+let addIsLoading = ref(false);
+let deleteIsLoading = ref(false);
+let completeIsLoading = ref(false);
+let editMode = ref(false);
+let newStepDescription = ref('');
+let steps = ref([]);
+let closeModal = ref(false);
+let stepToDelete = ref({});
+let currentlyLoadingStepId = ref(null);
+function isOwner() {
+  return web3Account.value === props.proposal.author;
+}
+function isComplete() {
+  return props.proposal?.state !== 'active';
+}
+function completedDate() {
+  return formatDate(new Date(props.proposal?.end * 1000));
+}
+function firstIncompleteStepId() {
+  return steps.value.find(step => step.stepStatus === 'active').id;
+}
+async function requireSignature() {
+  const storedAuth = localStorage.getItem('snap-progress');
+  if (storedAuth) {
+    return storedAuth;
+  }
+
+  const auth = getInstance();
+  let signature = await signMessage(
+    auth.web3,
+    this.$t('progress.confirmSignature'),
+    web3Account.value
+  );
+
+  const authHeader = `${props.proposal.id}-${signature}`;
+  localStorage.setItem('snap-progress', authHeader);
+
+  return authHeader;
+}
+async function getActiveSteps() {
+  if (isComplete) {
+    const apiUrl = `https://jissr670k3.execute-api.us-east-1.amazonaws.com/dev/proposal/${props.proposal.id}`;
+    try {
+      fetch(apiUrl).then(response =>
+        response
+          .json()
+          .then(
+            data =>
+              (steps.value = data.Items.sort((a, b) =>
+                a.index > b.index ? 1 : -1
+              ))
+          )
+      );
+    } catch (e) {
+      notify(['red', this.$t('progress.wentWrong')]);
+    }
+  }
+}
+async function createNewStep() {
+  if (newStepDescription.value !== '') {
+    try {
+      const sig = await requireSignature();
+      addIsLoading.value = true;
+      const apiUrl = `https://jci7szds71.execute-api.us-east-1.amazonaws.com/dev/proposal/${props.proposal.id}`;
+      const requestOptions = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: sig
+        },
+        body: JSON.stringify({
+          description: newStepDescription.value
+        })
+      };
+      fetch(apiUrl, requestOptions).then(response =>
+        response.json().then(() => {
+          getActiveSteps();
+          addIsLoading.value = false;
+          newStepDescription.value = '';
+        })
+      );
+    } catch (e) {
+      addIsLoading.value = false;
+      notify(['red', this.$t('progress.wentWrong')]);
```

Comment:
Getting errors on this "Cannot read properties of undefined (reading '$t')" and other instances of similar code. Investigating